python/iceberg/core/filesystem/filesystem_tables.py

```diff
@@ -37,7 +37,19 @@ def load(self: "FilesystemTables", table_identifier: str) -> Table:
 
     def create(self: "FilesystemTables", schema: Schema, table_identifier: str, spec: PartitionSpec = None,
                properties: dict = None, location: str = None) -> Table:
+        """
+        Create a new table on the filesystem.
+
+        Note: it is expected that the filesystem has atomic operations to ensure consistency for metadata updates.
+        Filesystems that don't have this guarantee could lead to data loss.
+
+        Only table_identifier or location should be not None. Will throw an error if both are not None.
```

Comment:
This doesn't agree with what is enforced. I like that `location` must always be `None` in the code so it would be good to state that here. It would also be easier to read since it avoids "should be not None".

Reply:
yup, apologies. Changed my mind halfway through impl. Fixed now